app/code/Mykhailok/SupportChat/Controller/Message/Save.php

```diff
@@ -3,91 +3,65 @@
 
-use Magento\Authorization\Model\UserContextInterface;
-use Magento\Framework\App\Action\Context;
-use Magento\Framework\Controller\Result\Json as JsonResult;
-use Magento\Framework\Controller\ResultFactory;
-use Magento\Framework\Exception\LocalizedException;
-use Mykhailok\SupportChat\Model\MessageUserDataProvider;
-use Mykhailok\SupportChat\Service\RequestValidate;
-use Mykhailok\SupportChat\Service\ResponseData;
-
-class Save extends \Magento\Framework\App\Action\Action implements
+class Save implements
+    \Magento\Framework\App\ActionInterface,
     \Magento\Framework\App\Action\HttpPostActionInterface
 {
-    /**
-     * @var \Mykhailok\SupportChat\Model\ChatMessageFactory
-     */
-    private $chatMessageFactory;
-
-    /**
-     * @var \Mykhailok\SupportChat\Model\MessageUserDataProviderFactory
-     */
-    private $messageUserDataProviderFactory;
-
-    /**
-     * @var \Mykhailok\SupportChat\Model\ResourceModel\ChatMessage
-     */
-    private $chatMessageResource;
-
-    /**
-     * @var \Magento\Store\Model\StoreManagerInterface
-     */
-    private $storeManager;
-
-    /**
-     * @var RequestValidate
-     */
-    private $requestValidate;
-
-    /**
-     * @var \Psr\Log\LoggerInterface
-     */
-    private $logger;
+    private \Mykhailok\SupportChat\Model\ChatMessageFactory $chatMessageFactory;
+    private \Mykhailok\SupportChat\Model\MessageUserDataProviderFactory $messageUserDataProviderFactory;
+    private \Mykhailok\SupportChat\Model\ResourceModel\ChatMessageFactory $chatMessageResourceFactory;
+    private \Mykhailok\SupportChat\Model\ResourceModel\ChatMessage $resourceModelChatMessage;
+    private \Magento\Store\Model\StoreManagerInterface $storeManager;
+    private \Mykhailok\SupportChat\Service\RequestValidate $requestValidate;
+    private \Magento\Framework\Message\ManagerInterface $messageManager;
+    private \Psr\Log\LoggerInterface $logger;
+    private \Magento\Framework\Controller\ResultFactory $resultFactory;
+    private \Magento\Framework\App\RequestInterface $request;
 
     public function __construct(
         \Mykhailok\SupportChat\Model\ChatMessageFactory $chatMessageFactory,
         \Mykhailok\SupportChat\Model\MessageUserDataProviderFactory $messageUserDataProviderFactory,
         \Mykhailok\SupportChat\Model\ResourceModel\ChatMessageFactory $chatMessageResourceFactory,
+        \Mykhailok\SupportChat\Model\ResourceModel\ChatMessage $resourceModelChatMessage,
         \Magento\Store\Model\StoreManagerInterface $storeManager,
         \Mykhailok\SupportChat\Service\RequestValidate $requestValidate,
         \Magento\Framework\Message\ManagerInterface $messageManager,
         \Psr\Log\LoggerInterface $logger,
-        Context $context
+        \Magento\Framework\Controller\ResultFactory $resultFactory,
+        \Magento\Framework\App\RequestInterface $request
     ) {
-        parent::__construct($context);
         $this->chatMessageFactory = $chatMessageFactory;
```

Comment:
@mykhailok-dv PHPStorm reports: `$this->chatMessageFactory is used only in constructor`
The same refers to `$this->chatMessageResourceFactory` (resource model is always a singleton, it does not require a factory anyway) and `$this->storeManager`